frontend/views/containers/chatroom/ChatMain.vue

```diff
@@ -858,23 +864,7 @@ export default ({
 
       if (!this.ephemeral.messagesInitiated) {
         this.setStartNewMessageIndex()
-
-        // NOTE: we do want the 'c-focused' animation if there is a message-scroll query.
-        if (events.length) {
-          // NOTE: if 'messageHashToScroll' was not there in the messages of the contract state
-          //       we need to retrieve more events, and render to scroll to that message
-          this.updateScroll(messageHashToScroll, Boolean(mhash))
-        } else {
-          // NOTE: we need to scroll to the message first in order to no more infiniteHandler is called
-          await this.updateScroll(messageHashToScroll, Boolean(mhash))
-
-          if (mhash) {
-            // NOTE: delete mhash in the query after scroll and highlight the message with mhash
-            const newQuery = { ...this.$route.query }
-            delete newQuery.mhash
-            this.$router.replace({ query: newQuery })
-          }
-        }
```

Comment:
This logic here handles scrolling to a particular message after message states are initialized. And moving this logic to below `this.ephemeral.messagesInitiated = true` is the fix for issue #2430 .
I noticed `infiniteHandler()` and `renderMoreMessages()` methods are called multiple times throughout the entire chatroom initialization process (which honestly I have not been able to figure out exactly from where and why..) and **observed** that executing scrolling logic here looked premature most of the time.
So moved this logic and it fixed the `mhash` issue.